### PR TITLE
Fix empty results on ExtractionFilter.

### PR DESCRIPTION
**Pull request by drcrallen**
* Now returns empty results rather than erroring out

Fixes 2 problems: 

1. If the dimension is not found in the segment this prevents an NPE
2. If there are no extraction matches in the segment this prevents an error in the OrFilter constructor.

### CODE REVIEW

**Comment by drcrallen** [comment]
@cheddar : AndFilter and OrFilter behave differently with empty filter lists. Word around the grapevine is that this may be some really old code. Do you happen to remember if there is a good reason to have the either refuse empty lists instead of just returning a blank ImmutableBitmap ?

File: processing/src/main/java/io/druid/segment/filter/ExtractionFilter.java [hunk 1]
```diff
@@ -66,6 +68,10 @@ public ExtractionFilter(
   @Override
   public ImmutableBitmap getBitmapIndex(BitmapIndexSelector selector)
   {
+    final List<Filter> filters = makeFilters(selector);
+    if (filters.isEmpty()) {
+      return new WrappedConciseBitmap();
+    }
     return new OrFilter(makeFilters(selector)).getBitmapIndex(selector);
```
Comment (xvrl) [inline processing/src/main/java/io/druid/segment/filter/ExtractionFilter.java]: it might make sense to fix OrFilter to take empty list instead, if that does not change the behavior.

Comment (xvrl) [inline processing/src/main/java/io/druid/segment/filter/ExtractionFilter.java]: let's add a test for empty filters

File: processing/src/main/java/io/druid/segment/filter/ExtractionFilter.java [hunk 2]
```diff
@@ -52,11 +53,12 @@ public ExtractionFilter(
   {
     final Indexed<String> allDimVals = selector.getDimensionValues(dimension);
     final List<Filter> filters = Lists.newArrayList();
-
-    for (int i = 0; i < allDimVals.size(); i++) {
-      String dimVal = allDimVals.get(i);
-      if (value.equals(fn.apply(dimVal))) {
-        filters.add(new SelectorFilter(dimension, dimVal));
+    if (allDimVals != null) {
```
Comment (xvrl) [inline processing/src/main/java/io/druid/segment/filter/ExtractionFilter.java]: let's add a test for missing dimension

File: processing/src/main/java/io/druid/segment/filter/ExtractionFilter.java (hunk 1 above)
Comment (drcrallen) [inline processing/src/main/java/io/druid/segment/filter/ExtractionFilter.java]: Added tests

File: processing/src/main/java/io/druid/segment/filter/ExtractionFilter.java (hunk 2 above)
Comment (drcrallen) [inline processing/src/main/java/io/druid/segment/filter/ExtractionFilter.java]: Added tests
